pymodbus/framer/socket_framer.py

```diff
@@ -36,13 +37,32 @@ class ModbusSocketFramer(ModbusFramer):
 
     method = "socket"
 
-    def __init__(self, decoder, client=None):
+    def __init__(self, decoder, **kwargs):
         """Initialize a new instance of the framer.
 
         :param decoder: The decoder factory implementation to use
         """
-        super().__init__(decoder, client)
+        super().__init__(decoder, **kwargs)
         self._hsize = 0x07
+        self.transport = kwargs.get('transport')
```

Comment:
If transport is need it should be an optional parameter, and not only in the socket framer, since most framers can be used with tcp.